app/models/azure_engine.rb

```diff
@@ -0,0 +1,154 @@
+require 'azure'
+require 'azure-contrib'
+require 'json'
+
+class AzureEngine
+  class Config
+    include ActiveSupport::Configurable
+    config_accessor :bucket_name
+    config_accessor :image_bucket_name
+    config_accessor :cdn_base_url
+    config_accessor :queue_name
+    config_accessor :azure_storage_access_key
+    config_accessor :azure_storage_account_name
+  end
+
+  def self.configure(&block)
+    yield config
+    Azure.config.storage_account_name = @config.azure_storage_account_name
+    Azure.config.storage_access_key = @config.azure_storage_access_key
+    @blob_client = Azure.blobs
+  end
+
+  def self.config
+    @config ||= Config.new
+  end
+
+  def self.queue_endpoint
+    url = "https://#{@config.azure_storage_account_name}.queue.core.windows.net/"
+    url
+  end
+
+  def self.resource_endpoint
+    unless @config.cdn_base_url.blank?
```

Comment:
__[rubocop]__
* Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
* Style/UnlessElse: Do not use `unless` with `else`. Rewrite these with the positive case first. [:ref](https://github.com/bbatsov/ruby-style-guide#no-else-with-unless)